modal/_container_entrypoint.py

```diff
@@ -740,6 +762,12 @@ def main(container_args: api_pb2.ContainerArguments, client: Client):
         else:
             run_with_signal_handler(call_function_async(function_io_manager, imp_fun))
 
+        # Commit on exit to catch uncommitted volume changes and surface background
+        # commit errors.
+        function_io_manager.volume_commit(
```

Comment:
Is it possible that the user code keeps open handles to files on the volume when this code is called?

Say, if a user spawns a thread that keeps an open file handle in write mode. There's nothing preventing that, right? As is, we will not include that file in the commit (unless you explicitly fsync it first)

Reply:
To get around that, we could extend `VolumeCommitRequest` with a flag to get include everything in the snapshot though, including files with active writers.

Reply:
Yes that is possible. I think implementing this in such a way that no files can be left open and all dirty pages are synced involves the more challenging 'on shutdown' implementation in the worker.

The quickest fix is https://docs.python.org/3/library/os.html#os.sync 

Seems to work on gvisor. These are produced in the fs on `os.sync()` for two open files:

<kbd>
<img width="357" alt="image" src="https://github.com/modal-labs/modal-client/assets/12058921/dc2abce9-b0f7-49de-ba73-ee07bd831cb6">
</kbd>

Reply:
We get `fsync`s after `os.sync()`? I did not expect that - I don't see that without gVisor. Interesting.